common/flatpak-dir.c

```diff
@@ -7673,40 +7725,27 @@ flatpak_dir_find_remote_ref (FlatpakDir   *self,
                                      &remote_refs, cancellable, error))
     return NULL;
 
-  remote_ref = find_matching_ref (remote_refs, name, opt_branch, opt_default_branch,
-                                  opt_arch, kinds, &my_error);
-  if (remote_ref == NULL)
+  remote_ref = find_ref_for_refs_set (remote_refs, name, opt_branch,
+                                      opt_default_branch, opt_arch, kinds,
+                                      out_kind, cancellable, &my_error);
+  if (!remote_ref)
     {
       if (g_error_matches (my_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
-        g_clear_error (&my_error);
-      else
         {
-          g_propagate_error (error, g_steal_pointer (&my_error));
+          g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                       _("Error searching remote %s: %s"),
```

Comment:
Is it worth it to rewrite this error message again? It's already manually written in `find_ref_for_refs_set()`

Reply:
Kind of - the intention here was to distinguish between a local repo and a remote